Dynamic Programming/Easy/Dice Combinations/Solution.md

```diff
@@ -0,0 +1,19 @@
+### Problem: [Dice Combinations](https://cses.fi/problemset/task/1633)
+### Topic: Dynamic Programming
+
+
+__Problem Statement:__ Your task is to count the number of ways to construct sum n by throwing a dice one or more times. Each throw produces an outcome between 1 and 6.
+
+To arrive at the step N, our previous one was N-1, N-2 or N-3....N-6, because we can only only get  1, 2 ... 6 from the dice.
+So, the number of ways we can get to N, is equal to the sum of ways we can go to N-1, N-2 to N-6 .
+
+```
+T(n):
+  if(n==0||n==1) return 1;
+  if(n<0) return 0;
+
+  return T(n-1) + T(n-2) + T(n-3) + T(n-4) + T(n-5) + T(n-6)
+```
+
+The above solution without memoitation has a time complexity of O(n^6) but reduces to Θ(n*6) when memotized!!!
+So it can be compiled before 1 sec.
```

Comment:
We are almost done with the review :)
I think that when using memoization the complexity drops to O(n)! Can you replace lines 19,20 with the following statement?:
"The above solution has a time complexity of O(n^6) but reduces to O(n) when applying memoization!"

Reply:
Right The number of operations becomes 6*n as there are 6 faces to the dice.
But the time complexity will be generalized as O(n). 

Done Thanks. :)